lib/pure/strformat.nim

```diff
@@ -226,6 +226,71 @@ template callFormatOption(res, arg, option) {.dirty.} =
   else:
     format($arg, option, res)
 
+proc impl(pattern: NimNode, shouldUnescape: bool): NimNode =
+  template makeStrLit(): NimNode =
+    if shouldUnescape:
+      newLit(unescape(strlit, prefix="", suffix=""))
+    else:
+      newLit(strlit)
+
+  if pattern.kind notin {nnkStrLit..nnkTripleStrLit}:
+    error "% only works with string literals", pattern
+  let f = pattern.strVal
+  var i = 0
+  let res = genSym(nskVar, "fmtRes")
+  result = newNimNode(nnkStmtListExpr, lineInfoFrom=pattern)
+  result.add newVarStmt(res, newCall(bindSym"newStringOfCap", newLit(f.len + count(f, '{')*10)))
+  var strlit = ""
+  while i < f.len:
+    if f[i] == '{':
+      inc i
+      if f[i] == '{':
+        inc i
+        strlit.add '{'
+      else:
+        if strlit.len > 0:
+          result.add newCall(bindSym"add", res, makeStrLit())
+          strlit = ""
+
+        var subexpr = ""
+        while i < f.len and f[i] != '}' and f[i] != ':':
+          subexpr.add f[i]
+          inc i
+        if i == f.len:
```

Comment:
This check was added in commit 2 to improve the error message when " is used incorrectly